File: Application Prorams/CPP/single.cpp
#include<iostream>
using namespace std;
class Demo //Parent class
{
    public:
        int x,y;

        Demo()
        {
            cout<<"1. Inside Demo constructor"<<endl;
            x=10;
            y=20;
        }

        ~Demo()
        {
            cout<<"1. Inside Demo destructor"<<endl;
        }

        void fun()
        {
            cout<<"Inside fun of Demo"<<endl;
        }
};

//child class:Hello
//class Hello extends Demo --java syntax
class Hello : public Demo
{
    public:
        int a,b,c;
    
    Hello()
    {
        cout<<"2. Inside Hello constructor"<<endl;
        a=30;
        b=40;
        c=50; 
    }

    ~Hello()
    {
        cout<<"2. Inside Hello destructor"<<endl;
    }

    void gun()
    {
        cout<<"Inside gun of Hello"<<endl;
    }
};

int main()
{
    Hello hobj;

    cout<<sizeof(hobj)<<"bytes"<<endl; //20 bytes

    cout<<hobj.x <<endl; //10
    cout<<hobj.y <<endl; //20
    cout<<hobj.a <<endl; //30
    cout<<hobj.b <<endl; //40
    cout<<hobj.c <<endl; //50

    hobj.fun();
    hobj.gun();
    
    return 0;
}

File: Application Prorams/JAVA/Single.java
class Base // parent
{
    public int A,B; //characteristics

    public Base()   //default constructor
    {
        System.out.println("Inside Base default constructor");
    }
    public void fun(){System.out.println("Inside base fun"); }

}

class Derived extends Base //derived is child
{
    public int P,Q;     //characteristics
    public Derived()    //default constructor
    {
        System.out.println("Inside Derived default constructor");
    }

    public void gun(){System.out.println("Inside Derived gun"); }
}

public class Single {
    public static void main(String[] args) {
        Derived dobj = new Derived(); //no casting
        System.out.println("Value of Base A "+dobj.A);
        dobj.fun();
        dobj.gun();
    }
}
